gremlin-dotnet/src/Gremlin.Net/Driver/ConnectionPool.cs

```diff
@@ -66,41 +61,45 @@ public async Task<IConnection> GetAvailableConnectionAsync()
 
         private async Task EnsurePoolIsPopulatedAsync()
         {
-            // The pool could have been empty because of connection problems. So, we need to populate it again.
-            while (true)
+            // The pool could have been (partially) empty because of connection problems. So, we need to populate it again.
+            while (_poolSize < NrConnections)
```

Comment:
Now that I think of it, maybe it would be safer to replace the `while` with `if` statement, wdyt?

Reply:
It's probably clearer like that, yes. With an `if` it won't really work any more when the pool is completely empty and two requests come in in parallel, but letting the second request loop until the first has populated the pool is probably not a really good idea in general. With an `if` we will just fail fast here.
Until we have a mechanism to create connections in the background again with a timer or something like that, failing fast is probably the better and safer option here.